libsql-server/src/connection/libsql.rs

```diff
@@ -391,14 +393,43 @@ where
         ctx: RequestContext,
         builder: B,
     ) -> Result<(B, Program)> {
+        struct Bomb {
+            canceled: Arc<AtomicBool>,
+            defused: bool,
+        }
+
+        impl Drop for Bomb {
+            fn drop(&mut self) {
+                if !self.defused {
+                    tracing::trace!("cancelling request");
+                    self.canceled.store(true, Ordering::Relaxed);
+                }
+            }
+        }
+
+        let canceled = {
+            let cancelled = self.inner.lock().canceled.clone();
+            cancelled.store(false, Ordering::Relaxed);
+            cancelled
+        };
+
+        let mut bomb = Bomb {
+            canceled,
+            defused: false,
+        };
+
         PROGRAM_EXEC_COUNT.increment(1);
 
         check_program_auth(&ctx, &pgm, &self.inner.lock().config_store.get())?;
```

Comment:
Will cancellation be triggered if auth check will fail?

Reply:
Maybe better to put `bomb` after this check right before exec?

Reply:
The flag will be raised yes, but but won't execute the program so this won't do anything. What do you think could be an issue? Actually we probably never need to defuse the bomb, since we always re-initialize it.

Reply:
The only issue with this implementation is if 2 execute are performed concurrently on the same connection, but that isn't possible with hrana

Reply:
IIUC, in this case log `tracing::trace!("cancelling request");` will be produced which may be a bit misleading

Reply:
fair